### ISSUE TITLE
[bug] FMD 2 crashing while running idle checking manga in background

### ISSUE THREAD

**Issue by molitar** [dazedcat19/FMD2#93]
**Prerequisite**
- [x ] I'm using the latest version.
- [ ] All website modules are up to date.

- [x ] This is not temporary server issue.

**System information**
Windows 10 build 1909 64-bit

**Describe the bug**
Crash error while it was remaining idle overnight so it's set to check for new chapters every 4 hours.

**To Reproduce**
Not sure it happened overnight.

**Expected behavior**
Should remain up and not show an exception error.

**Additional context**
Program exception!
Application       : Free Manga Downloader
Version           : 2.0.23.0
Product Version   : Win64
Host Machine      : Windows 10 64-bit
FPC Version       : 3.3.1
LCL Version       : 2.1.0.0
WidgetSet         : Win32/Win64
Target CPU-OS     : x86_64-Win64
Build Time        : 2020/08/21 12:22:08
Path              : C:\DOS\fmd_2.0\fmd.exe
Process ID        : 69984
MainThread ID     : 67664
Thread ID         : 67664
Time              : 26/08/2020 03:26:37.733
Sender Class      : TFavoriteManager
Exception Object  : 000000001065CB00
Exception Class   : ESQLDatabaseError
Message           : TSQLite3ConnectionH : database schema has changed

**Comment by dazedcat19** [comment]
I need reproducible steps.

**Comment by molitar** [comment]
Only thing I did was let the screen sleep the computer stays awake.. It remains open all night and I wake up finding an exception error.  Also if I try to click on a manga and it is unable to find the chapters it will cause and exception error.  So maybe what is causing it is the site is not working and than it runs through check for chapters and errors out is my guess.. Especially when I have 380 manga in my list.

**Comment by molitar** [comment]
So it seems this happens when site goes down.. I forgot to uncheck Kissmanga and remove the only manga left on it so when it tried to update chapter from it.. it crashes. It happened when I had done a check for new chapters in favorites.

Program exception!
Application       : Free Manga Downloader
Version           : 2.0.23.0
Product Version   : Win64
Host Machine      : Windows 10 64-bit
FPC Version       : 3.3.1
LCL Version       : 2.1.0.0
WidgetSet         : Win32/Win64
Target CPU-OS     : x86_64-Win64
Build Time        : 2020/08/21 12:22:08
Path              : C:\DOS\fmd_2.0\fmd.exe
Process ID        : 17716
MainThread ID     : 24708
Thread ID         : 67720
Time              : 26/08/2020 21:56:30.780
Sender Class      : TGetMangaInfosThread
Exception Object  : 000000000F79F4B0
Exception Class   : EDatabaseError
Message           : Could not find the requested record.

**Comment by dazedcat19** [comment]
"database schema has changed" tell us there's an asynchronous operation to the database. I check the source code. favoritesmanager touching the database only when add/delete an item. restore/backup from database at init and exit, which should be in blocking mode. every favorites is loaded to the memory. so there's really not much operation with databases on runtime.

**Comment by molitar** [comment]
I figured out the problem.  The problem is the site has gone down it will than crash. Repeatable behavior for me which I did below.

1. Make sure manhuas is selected
2.Double click on any manhua like Cultivation Return on Campus
3. Crash 

Site seems to be in maintenance so it can not find any chapters and it crashes.  This is same problem when it is checking for chapters and site is down for either maintenance or some other reason.  I have found other chapters that has same problem when the chapters can not be found get this error.
